azure/install-rust.yml

```diff
@@ -52,5 +53,9 @@ steps:
       rustup --version
     displayName: Check installed rust version
 
+  # Checkout any submodules the repository may have
```

Comment:
Since this is a step template, why do we need to bake in `checkout`?  Can't that be left to the caller to specify `checkout`?

Reply:
for example, what if people need other `checkout` settings?  Would them specifying `checkout` and us specifying it cause problems?

Reply:
My concern here is that in the majority of jobs, you will want recursive submodule checkout. And since it has to be specified for _every_ job independently, that would lead to a lot of duplication. Every template we provide that uses `install-rust` would need to _also_ include a checkout step before each one, _and_ that would have to be optional there too in case the caller wants to use _that_ template with a different checkout.

My thinking here was that making this less configurable leads to both cleaner code and a better end-user experience (people would be surprised if their submodules weren't also checked out). You can always fall back to writing your own jobs if you specifically need a _different_ `checkout: self` setting. That said, we _could_ do something like have `submodules: null` remove the entire `checkout` step to allow people to re-use `install-rust` even if they want their own checkout. I'll file a PR for that!

Reply:
See https://github.com/crate-ci/azure-pipelines/pull/46

Reply:
I can understand the desire for the default but as a step we're providing in of itself, coupling `checkout` to this seems a bad mix.

Alternatively, we could have a "by convention" private step (leading `_` like in python) to house "common stuff we do between all job".

Reply:
I don't disagree. Let's continue discussion in #46.